<?xml version="1.0" ?>
<sdf version="1.5">
  <world name="default">
    <gui>
      <camera name="user_camera">
        <pose>28.124 -28.95 9.52 0 0.58 2.61</pose>
      </camera>
    </gui>

    <scene>
          <ambient>0.25 0.3 1 1</ambient>
          <grid>0</grid>
          <background>0.25 0.3 1 1</background>
          <shadows>1</shadows>
    </scene>

    <light type="directional" name="sun0">
      <cast_shadows>1</cast_shadows>
      <pose>8 -21 10 0 0.8 -2.8</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
    <light type="directional" name="sun1">
      <cast_shadows>1</cast_shadows>
      <pose>8 -20 10 0 0.8 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
    <light type="directional" name="sun2">
      <cast_shadows>1</cast_shadows>
      <pose>8 -19 10 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
    <light type="directional" name="sun3">
      <cast_shadows>1</cast_shadows>
      <pose>8 -19 10 1.75 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
    <light type="directional" name="sun4">
      <cast_shadows>1</cast_shadows>
      <pose>8 -19 10 1.75 0 2.9</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <model name="ceiling_plane">
      <static>true</static>
      <pose>0 0 4.49 0 0 0</pose>
      <link name="link">
        <!-- <collision name="collision"> -->
        <!--   <geometry> -->
        <!--     <box> -->
        <!--       <size>90 60 .001</size> -->
        <!--     </box> -->
        <!--   </geometry> -->
        <!-- </collision> -->
        <visual name="visual">
          <cast_shadows>false</cast_shadows>
          <geometry>
            <box>
              <size>90 60 .001</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/BlueTransparent</name>
            </script>
          </material>
          <transparency>0.6</transparency>
        </visual>
      </link>
    </model>

    <include>
      <uri>model://gibb</uri>
    </include>


    <model name="pinger_a">
      <static>true</static>
      <pose>37.88 -4.1 -4.9 0 0 0</pose>
      <link name="link">
        <visual name="visual">
          <cast_shadows>true</cast_shadows>
          <geometry>
            <cylinder>
                <radius> 0.03 </radius>
                <length> 0.15 </length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/FlatBlack</name>
            </script>
          </material>
        </visual>
      </link>
    </model>

    <include>
      <name>gate_side_a</name>
      <pose>19.8 -25.7 4.5 0 0 1.575</pose>
      <uri>model://gate</uri>
    </include>

    <include>
      <name>buoys</name>
      <pose>16.1 -25.5 1.55 0 0 1.575</pose>
      <uri>model://buoys</uri>
    </include>

    <include>
        <name>robosub</name>
        <pose>21 -25 3 0 0 3</pose>
        <uri>model://cobalt</uri>
    </include>

  </world>
</sdf>
